File: backend/app/models/job.py
from sqlmodel import SQLModel, Field
from datetime import datetime
from typing import Optional

class Job(SQLModel, table=True):
    id: Optional[int] = Field(default=None, primary_key=True)
    title: str
    company: str
    location: str
    description: str
    salary: Optional[str] = None
    job_url: str
    source: str = Field(default="indeed")  # indeed, linkedin, etc.
    scraped_at: datetime = Field(default_factory=datetime.utcnow)
    keywords: Optional[str] = None
    is_tracked: bool = Field(default=False)
    created_at: datetime = Field(default_factory=datetime.utcnow)
    updated_at: datetime = Field(default_factory=datetime.utcnow)

File: android/app/src/main/java/com/Vedang/careerflow/model/Job.kt
package com.Vedang.careerflow.model

/**
 * A job as the Android app uses it after it has been mapped from the API response.
 */
data class Job(
    val id: Int,
    val title: String,
    val company: String,
    val location: String,
    val description: String?,
    val salary: String?,
    val jobUrl: String,
    val source: String,
    val keywords: String?,
    val isTracked: Boolean,
    val scrapedAt: String?,
    val createdAt: String?,
    val updatedAt: String?
)
